rubin_scheduler/skybrightness_pre/sky_model_pre.py

```diff
@@ -280,14 +327,36 @@ def return_mags(
 
         # Check if we are between sunrise/set
         if baseline > self.timestep_max + 1e-6:
-            warnings.warn("Requested MJD between sunrise and sunset, returning closest maps")
-            diff = np.abs(self.mjds[left.max() : right.max() + 1] - mjd)
-            closest_indx = np.array([left, right])[np.where(diff == np.min(diff))].min()
-            sbs = {}
-            for filter_name in filters:
-                sbs[filter_name] = self.sb[filter_name][closest_indx, indx]
-                sbs[filter_name][np.isinf(sbs[filter_name])] = badval
-                sbs[filter_name][np.where(sbs[filter_name] == hp.UNSEEN)] = badval
+
+            # Check if sun is really high:
+            obstime = Time(mjd, format="mjd")
+            sun = get_sun(obstime)
+            aa = AltAz(location=self.location, obstime=obstime)
+            sun_alt_az = sun.transform_to(aa)
+
+            if sun_alt_az.alt.rad > self.sun_alt_limit:
+                warnings.warn("Sun high, using bright sky approx")
+                hp_aa = self.skycoord.transform_to(aa)
+
+                sbs = {}
+                for filter_name in filters:
+                    sbs[filter_name] = simple_daytime(
+                        hp_aa.alt.deg,
+                        hp_aa.az.deg,
+                        sun_alt_az.alt.deg,
+                        sun_alt_az.az.deg,
+                        filter_name=filter_name,
+                    )
+
+            else:
+                warnings.warn("Requested MJD between sunrise and sunset, returning closest maps")
```

Comment:
This looks fine, but a next level more sophisticated (but still pretty poor) would be to keep the map in alt/az space rather than ra/dec space, so part of the sky covered by the map doesn't slip off of the zenith distance range accessible to the telescope.